lib/rules/no-computed-in-data.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @fileoverview Ensure computed properties are not used in the data()
+ * @author IWANABETHATGUY
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+const utils = require('../utils')
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'require computed properties are not used in the data()',
+      categories: ['strongly-recommended'],
+      recommended: false,
+      url: 'https://eslint.vuejs.org/rules/no-computed-in-data.html'
+    },
+    fixable: null, // or "code" or "whitespace"
+    schema: [
+      // fill in your schema
+    ]
+  },
+
+  create: function (context) {
+    let dataAstNode
+    const memberExpressionMap = Object.create(null)
+    return {
+      ObjectExpression (node) {
+        if (!dataAstNode) {
+          dataAstNode = node.properties.find(
+            p =>
+              p.type === 'Property' &&
+              p.key.type === 'Identifier' &&
+              p.key.name === 'data'
+          )
+        }
+      },
+      'Property:exit' (node) {
+        if (node === dataAstNode) {
+          dataAstNode = undefined
+        }
+      },
+      MemberExpression (node) {
+        if (dataAstNode) {
+          // a memberExpression `like this.a.c.d` -> when traverse to c.d we can got the the full name -> this.a.c.d
+          const fullName = utils.parseMemberExpression(node).join('.')
```

Comment:
I think it's enough to check the first two.

```suggestion
          const fullName = utils.parseMemberExpression(node).slice(0,2).join('.')
```

I think this change eliminates the need to use startsWith. at 61.